src/components/MerchantsPage/gam/CampaignProgressBar.tsx

```diff
@@ -0,0 +1,135 @@
+import * as React from 'react';
+import styled from 'styled-components';
+
+interface Props {
+  numContributions: number;
+  targetAmount: number;
+  progressBarColor: string;
+  lastContributionTime: Date;
+  endDate: Date;
+}
+
+const CampaignProgressBar = ({
+  numContributions,
+  targetAmount,
+  progressBarColor,
+  lastContributionTime,
+  endDate,
+}: Props) => {
+  const timeSinceLastContribution = Date.now() - lastContributionTime.getTime();
+  const minutesSinceLastConstribution = timeSinceLastContribution / (1000 * 60);
+  const hoursSinceLastContribution = minutesSinceLastConstribution / 60;
+  const daysSinceLastContribution = hoursSinceLastContribution / 24;
+
+  const progressWidth = (raised: number, target: number) => {
+    if (raised < target) return (raised / target) * 100;
+    return 100;
+  };
+  const progressPercent = Math.floor((numContributions / targetAmount) * 100);
+
+  const timeUntilEnd = endDate.getTime() - Date.now();
+  const daysUntilEnd = timeUntilEnd / (1000 * 3600 * 24);
+
+  return (
+    <ProgressBarContainer>
+      {hoursSinceLastContribution < 1 && (
+        <TimeStamp>
+          Last contribution made {Math.floor(minutesSinceLastConstribution)}m
+          ago
+        </TimeStamp>
+      )}
+      {hoursSinceLastContribution >= 1 && hoursSinceLastContribution < 24 && (
+        <TimeStamp>
+          Last contribution made {Math.floor(hoursSinceLastContribution)}h ago
+        </TimeStamp>
+      )}
+      {hoursSinceLastContribution >= 24 && (
+        <TimeStamp>
+          Last contribution made {Math.floor(daysSinceLastContribution)}d ago
+        </TimeStamp>
+      )}
+      <TargetAmountBar className="progress-bar">
+        <CurrentProgressBar
+          style={{
+            width: `${progressWidth(numContributions, targetAmount)}%`,
+            backgroundColor: progressBarColor,
+            //defaults to default color if no color is passed in
+          }}
+        >
+          {' '}
```

Comment:
I've noticed you've got a few of these. Maybe something to better account for in margin/padding?

Reply:
Ahh yeah I copied this alot of this code over from ProgressBar. Will remove them